### PR TITLE
PP-2790 Backfill ChargeTransactions with charges from 17-11-2017

### PR DESCRIPTION
**Pull request by georgeracu**
- Update transactions status during window of 3-4pm on 17-11-2017

🎅

### CODE REVIEW

**Review by alexbishop1 — CHANGES_REQUESTED** [review]
This doesn’t look like it’s what we want: there is already a transaction for each of the affected payments, it just has the wrong status. We want to be doing updates, not inserts.

File: src/main/resources/migrations.xml [hunk 1]
```diff
@@ -1075,5 +1075,35 @@
...
+                REPLACE(ch.status, ' ', '_'),
+                'CHARGE'
+            -- use all PaymentRequest
+            FROM payment_requests pr
+            -- use all ChargeTransaction and fill missing one from PaymentRequest with null
+            LEFT JOIN transactions t ON pr.id = t.payment_request_id AND t.operation='CHARGE'
+            -- use only Charge that has a PaymentRequest
+            JOIN charges ch ON pr.external_id = ch.external_id
+            -- use only PaymentRequest that don't have a ChargeTransaction
+            WHERE t.id IS NULL
```
Comment (chrisgrimble) [inline src/main/resources/migrations.xml]: Don't these payments have a charge transaction thought?

**Review by alexbishop1 — DISMISSED** [review]
Not thrilled with the name `a` but it looks like it will work 🎄

**Review by alexbishop1 — APPROVED** [review]
☃️

**Pull request merged**